### PR TITLE
Support matrix

### PR DESCRIPTION
**Pull request by dbabokin**
* Print support matrix under --support-matrix
* Use information about dynamically link targets to report supported targets/arches/oses.

### CODE REVIEW

File: src/target_enums.cpp [hunk 1]
```diff
@@ -332,6 +332,26 @@ std::string OSToString(TargetOS os) {
     UNREACHABLE();
 }
 
+std::string OSToLowerString(TargetOS os) {
+    switch (os) {
+    case TargetOS::windows:
+        return "window";
```
Comment (aneshlya) [inline src/target_enums.cpp]: Please fix "window"->"windows"

File: src/target_registry.cpp [hunk 2]
```diff
@@ -135,3 +140,143 @@ const BitcodeLib *TargetLibRegistry::getISPCTargetLib(ISPCTarget target, TargetO
...
+    // OS names row
+    std::vector<std::string> os_names;
+    os_names.push_back("");
+    for (int j = (int)TargetOS::windows; j < (int)TargetOS::error; j++) {
+        os_names.push_back(OSToString((TargetOS)j));
+    }
+    table.push_back(os_names);
+
+    // Fill in the name, one target per the row.
+    for (int i = (int)ISPCTarget::sse2_i32x4; i < (int)ISPCTarget::error; i++) {
```
Comment (aneshlya) [inline src/target_registry.cpp]: Do we have some guarantee that we will not change the order of targets in enums and sse2_i32x4 will be always the 1st one?
Maybe it makes sense to have "none" as the 1st value for targets and OSes enums if it does not introduce many changes.

File: src/target_registry.cpp [hunk 3]
```diff
@@ -135,3 +140,143 @@ const BitcodeLib *TargetLibRegistry::getISPCTargetLib(ISPCTarget target, TargetO
...
+                    arch_list_os += ArchToString(arch);
+                }
+            }
+            arch_list_target.push_back(arch_list_os);
+            row.push_back(arch_list_os);
+        }
+        table.push_back(row);
+    }
+
+    // Colect maximum sizes for all columns
```
Comment (aneshlya) [inline src/target_registry.cpp]: Colect->collect

File: src/target_registry.cpp (hunk 2 above)
Comment (dbabokin) [reply src/target_registry.cpp]: Iterating through enum values is ugly. I suggest keeping it this way. I we have more places, where we have to iterate through these enums, we'll need to come up with a better way.

File: src/target_registry.cpp (hunk 3 above)
Comment (dbabokin) [reply src/target_registry.cpp]: Fixed.

File: src/target_enums.cpp (hunk 1 above)
Comment (dbabokin) [reply src/target_enums.cpp]: That's really good catch. Thanks! Fixed.

**Pull request merged**